UI/timeline2

cconard96 opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

orthagh commented:
Some issues i found while testing : 

 - Some fields in task doesn't seems to have `'icon_label': true,` option like `state`, `duration`  and it broke alignment of the form
![image](https://user-images.githubusercontent.com/418844/112135055-a81d9680-8bcd-11eb-918b-9065656800f5.png)

- A class is missing when adding a new task: the yellow background is missing
![image](https://user-images.githubusercontent.com/418844/112135287-ea46d800-8bcd-11eb-94aa-8c2a49740dbb.png)

- task.duration icon is not the good one, it should be `fa-stopwatch`
![image](https://user-images.githubusercontent.com/418844/112135516-27ab6580-8bce-11eb-97e1-f66c4afd2db0.png)

- for user mention feature, we miss the css. see : https://github.com/glpi-project/glpi/pull/8751/files#diff-76628e774773cc79d896ef83f3617a00239f1a6a477a0b537f6612de5dc7e4a6R350-R361
  The css rule should be moved into a separate file included in `css/style.scss`

- the can add document should be removed completely i think, we though the "add document" action is useless. (Display/"edit of existing" actions should be kept)
  
- there is on the `feat/modern-ui` branch deprecated calls of `Html::setRichTextContent` polluting the display of timeline. I asked @cedric-anne to fix this.

-  Maybe due to the last point, i couldn't check display of documents in timeline. But a rapid read shows you still have raw code 
 (`Sample Image.png`, `Screenshot1.png`)

cconard96 commented:
> * Some fields in task doesn't seems to have `'icon_label': true,` option like `state`, `duration`  and it broke alignment of the form
Fixed

> * A class is missing when adding a new task: the yellow background is missing
Fixed. The CSS uses "ITILTask" so when I added the Timeline feature and switched to using the actual class "CommonITILTask", it broke.

> * for user mention feature, we miss the css. see : https://github.com/glpi-project/glpi/pull/8751/files#diff-76628e774773cc79d896ef83f3617a00239f1a6a477a0b537f6612de5dc7e4a6R350-R361
I just started adding this and then realized it is already included in `css/includes/components/richtext.scss`

> * the can add document should be removed completely i think, we though the "add document" action is useless. (Display/"edit of existing" actions should be kept)
I'm not sure what you mean about this.

> * i couldn't check display of documents in timeline. But a rapid read shows you still have raw code
That template wasn't meant to be included. I was prototyping UX for adding new documents to the timeline.
